aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -972,19 +1038,65 @@ public void connect(
             final String customAuthorizer,
             final AWSIotMqttClientStatusCallback statusCallback) {
 
-        if (tokenKeyName == null || token == null || tokenSignature == null) {
+        if (tokenKeyName == null || token == null) {
             throw new IllegalArgumentException("tokenKeyName/token/tokenSignature cannot be null");
         }
 
-        this.userStatusCallback = statusCallback;
+        userStatusCallback = statusCallback;
+        authMode = AuthenticationMode.CUSTOM_AUTH;
+
+        // Set custom websocket header
+        customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Name", customAuthorizer);
+        customWebsocketHeaders.setProperty(tokenKeyName, token);
+        if (tokenSignature != null) {
+            customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Signature", tokenSignature);
+        }
+        customAuthConnect(getOptions());
+    }
+
+    /**
+     * Initializes the MQTT session and connects to the specified MQTT server
+     * using username and password as credentials.
+     *
+     * @param username Username to authenticate client to the broker.
+     * @param password Password to authenticate client to the broker.
+     * @param statusCallback When a new MQTT session status is received, one of the methods inside this
+     *                       callback will be invoked with new connection status.
+     */
+    public void connect(
+            final String username,
+            final String password,
+            final AWSIotMqttClientStatusCallback statusCallback) {
+
+        if (username == null || password == null) {
+            throw new IllegalArgumentException("username and/or password cannot be null.");
+        }
 
+        this.username = username;
+        this.password = password;
+        userStatusCallback = statusCallback;
+        authMode = AuthenticationMode.USERNAME_PASSWORD;
+        customAuthConnect(getOptions());
+    }
+
+    private MqttConnectOptions getOptions() {
```

Comment:
`getFoo()` sounds like an accessor to a cached value, this  should be named like a factory method, i.e., `createOptions()`, or `buildOptions()`, or `generateOptions()` whatever.